tests/unit/test_alias.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from cromshell.alias import command as alias_command
+
+
+class TestAlias:
+    """Test the alias command functions"""
+
+    @pytest.mark.parametrize(
+        "alias_name, alias_validity",
+        [
+            ["goodalias", True],
+            ["-badalias", False],
+            ["good-alias", True],
+            ["goodalias-", True],
+            ["bad alias", False],
+            ["321", False],
+        ],
+    )
+    def test_alias_is_valid(self, alias_name: str, alias_validity: bool) -> None:
+        assert alias_command.alias_is_valid(alias_name=alias_name) is alias_validity
+
+    @pytest.mark.parametrize(
+        "alias_name, alias_existence",
+        [
+            ["wombat", True],
+            ["foo", False],
+            ["Failed", False],
+        ],
+    )
+    def test_alias_exists(
+        self,
+        alias_name: str,
+        alias_existence: bool,
+        mock_workflow_database_tsv,
+    ) -> None:
+        assert (
+            alias_command.alias_exists(
+                alias_name=alias_name, submission_file=str(mock_workflow_database_tsv)
+            )
+            is alias_existence
+        )
+
+    @pytest.mark.parametrize(
+        "workflow_id, alias_existence",
+        [
+            ["a63aa10c-a43e-4ca7-9be9-c2d2aa08b96d", True],
+            ["b3b197b3-fdca-4647-9fd8-bf16d2cb734d", False],
+        ],
```

Comment:
again, seems like the -1, -2 use cases are not covered.

Reply:
add to correct test file for function testing

Reply:
In the alias command script the workflow id is resolved by another function if given a relative id then passed to this function. This function only excepts workflow_id which is specified in the description "workflow_id: Hexadecimal identifier of workflow submission"
workflow-id OR relative-id -> **Function to Resolve workflow id** -> workflow-id -> **Function Check Workflow Has Alias**